modules/repository-datastax/src/main/java/com/intuit/wasabi/repository/cassandra/UninterruptibleUtil.java

```diff
@@ -0,0 +1,51 @@
+package com.intuit.wasabi.repository.cassandra;
+
+import com.datastax.driver.core.exceptions.DriverException;
+import com.datastax.driver.core.exceptions.DriverInternalError;
+import com.datastax.driver.mapping.Result;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.Uninterruptibles;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Created by nbarge on 1/4/17.
+ */
+public class UninterruptibleUtil {
+
+    public static <T> Result<T> getUninterruptibly(ListenableFuture<Result<T>> aFuture, long timeout, TimeUnit unit) throws TimeoutException {
```

Comment:
Guava is great, no question about it, and this wrapper already uses Guava's utility to get result from future (#1).. But Guava alone is not enough as we still need this wrapper to convert ExecutionException to DATASTAX specific more meaningful exception (#2)